R/readBuell2011.R

```diff
@@ -0,0 +1,47 @@
+---
+#' Raw read for USGS_S3C (2011)
+#' 
+#' Download and read in raw data from Myers-Smith, Isla. 2005. Soil data for cores from a transect from the center of the BBC collapse scar into the surrounding burn, Bonanza Creek LTER - University of Alaska Fairbanks. BNZ:192, http://www.lter.uaf.edu/data/data-detail/id/192. doi:10.6073/pasta/b0e9120983438c27bf1a30f37f0e5200
+#' 
+#' This is not post-processed or QA/QCed by this package.
+#'
+#' @param dataDir filename for download directory
+#' @param download boolean that will download the files from repository
+#'
+#' @return a data frame containing newly ingested data from the Myers-Smith dataset
+#' 
+#' @export
+
+#' @importFrom readr read_delim
+#'
+readBuell2011 <- function(dataDir, download=TRUE, verbose=FALSE){
+  #### Download ####
+  
+  BuellDownload_url <- 'https://pubs.usgs.gov/of/2004/1227/s3c_ascii.zip'
+  BuellDownload_target <- file.path(dataDir, 's3c_ascii.zip') 
+  
+  BuellDataFiles <- file.path(dataDir, 's3c_ascii',
+                             c("horiz01.txt","pedon01.txt", "taxon01.txt"))
+  
+  if(!all(file.exists(BuellDataFiles))){
```

Comment:
You want to check that the target files are downloaded (in this case the zip file) not the data files.

Reply:
Also think about if the `all` function is needed here. If it's only one file you are downloading then you don't need to add `all` to the boolean evaluation.